frontend/front/src/pages/Surveyor/account/Account.js

```diff
@@ -3,42 +3,69 @@ import React from "react";
 import AccountDetail from "./AccountDetail";
 import { Link } from "react-router-dom";
 import { useSelector } from "react-redux";
+import { selectCurrentUser } from "../../../features/login/loginSlice";
+import { useGetSurveyorQuery } from "../../../api/apiSlice";
+import Loader from "../../../components/Loader";
+import CustomSnackbar from "../../../components/CustomSnackbar";
 
 const Account = () => {
-  const { firstName, lastName, email, address, phoneNumber } = useSelector(
-    (state) => state.account
-  );
+  const { id: userId } = useSelector(selectCurrentUser);
+
+  const {
+    data: {
+      firstname: firstName,
+      lastname: lastName,
+      email,
+      street_address: address,
+      phone: phoneNumber,
+    },
+    isError: isAccountDataError,
+    isLoading: isAccountDataLoading,
+  } = useGetSurveyorQuery(userId);
+
   return (
     <Box>
-      <Grid container direction="column" rowSpacing={1}>
-        <Grid item xs={12}>
-          <Box display="flex" px={2} pt={3}>
-            <Typography variant="h2" component="h2">
-              My Account
-            </Typography>
-          </Box>
-        </Grid>
-        <AccountDetail label="First Name" value={firstName} />
-        <AccountDetail label="Last Name" value={lastName} />
-        <AccountDetail label="Email" value={email} />
-        <AccountDetail label="Address" value={address} />
-        <AccountDetail label="Phone Number" value={phoneNumber} />
-      </Grid>
-      <Grid container direction="column" rowSpacing={4}>
-        <Grid item xs={12}>
-          <Box display="flex" mt={3} mb={2} px={2}>
-            <Button variant="contained" component={Link} to="edit">
-              Edit
-            </Button>
-          </Box>
-          <Box p={1} sx={{ borderBottom: "1px dashed grey" }} />
-          <Box display="flex" mt={3} mb={2} px={2}>
-            <Button variant="contained" disabled={true}>
-              Request New Assignment
-            </Button>
-          </Box>
-        </Grid>
-      </Grid>
+      {isAccountDataLoading ? (
+        <Loader />
+      ) : isAccountDataError ? (
+        <CustomSnackbar
+          open={isAccountDataError}
+          message="Error fetching Surveyors data."
```

Comment:
"Surveyors" or "Surveyor's"? (or "surveyor"?)

Reply:
Probably "account data" is better in the context of this page. Copy and pasted this from another component and forgot to fix that detail!